core/templates/pages/exploration-editor-page/services/user-exploration-permissions.service.ts

```diff
@@ -29,19 +30,33 @@ import { ExplorationPermissions } from
   providedIn: 'root'
 })
 export class UserExplorationPermissionsService {
+  private userExplorationPermissionsFetched = new EventEmitter<void>();
+
   constructor(
     private explorationPermissionsBackendApiService:
-    ExplorationPermissionsBackendApiService) {}
+    ExplorationPermissionsBackendApiService) {
+  }
 
   static permissionsPromise: Promise<ExplorationPermissions> = null;
 
   async getPermissionsAsync(): Promise<ExplorationPermissions> {
     if (!UserExplorationPermissionsService.permissionsPromise) {
       UserExplorationPermissionsService.permissionsPromise = (
-        this.explorationPermissionsBackendApiService.getPermissionsAsync());
+        this.fetchPermissionsAsync());
     }
     return UserExplorationPermissionsService.permissionsPromise;
   }
+
+  async fetchPermissionsAsync(): Promise<ExplorationPermissions> {
+    UserExplorationPermissionsService.permissionsPromise = (
+      this.explorationPermissionsBackendApiService.getPermissionsAsync());
+    this.userExplorationPermissionsFetched.emit();
```

Comment:
Do we need to emit this only after the `getPermissionsAsync` function completes i.e, after getting response from the server?

Reply:
Also, are we expecting to `fetch` function to make HTTP call and `get` method to use `fetch` function if the response is not cached?

Reply:
> Do we need to emit this only after the getPermissionsAsync function completes i.e, after getting response from the server?

Yes, only after the user permission gets cached should the `emit` run for other functions to get the user exploration permissions from the cache.
After `getPermissionsAsync` function runs the user exploration permissions gets cached.

> Also, are we expecting to fetch function to make HTTP call and get method to use fetch function if the response is not cached?

Yes, I did that to make the code simpler. So that the `http` call is done from one place of the code.

Reply:
> Yes, only after the user permission gets cached should the emit run for other functions to get the user exploration permissions from the cache.

If I'm reading the code correctly then I don't think the `emit` is happening after the HTTP request is completed. I think it's happening in sync.

Reply:
@DubeySandeep 
Shall I change the code to this:
```
    UserExplorationPermissionsService.permissionsPromise = (
     await this.explorationPermissionsBackendApiService.getPermissionsAsync());
```

Reply:
Nope, I was thinking something around:
this.explorationPermissionsBackendApiService.getPermissionsAsync().then(function() {
    this.userExplorationPermissionsFetched.emit();
})

Are we expecting the emit to happen in the promise call?

Reply:
DONE